### PR TITLE
Deprecate unused `wazuh-db` code

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#12356|

## Description

This PR deprecates unused Wazuh-DB helpers, functions, and queries. 

**Note: The list detailed below implies also the removal of other related functions, wrappers, UT, and wazuh-db commands**

### Wazuh-db commands

- "vuln_cves clear"
- "set-labels"
- "delete-agent-belong"
- "select-keepalive"

### wdb_agents_helpers.c

- wdb_clear_vuln_cves()
- wdb_remove_vuln_cves()

### wdb_global_helpers.c

- wdb_set_agent_labels()
- wdb_get_agent_keepalive()
- wdb_delete_agent_belongs()
- wdb_select_group_belong()

### wdb_fim.c

- wdb_get_last_fim()
- wdb_insert_fim()
- wdb_find_file()
- wdb_insert_file()
- wdb_delete_fim_all()
- wdb_delete_fim()

### wdb_integrity.c

- wdbi_array_hash()

### wdb_metadata.c

- wdb_fim_fill_metadata()
- wdb_metadata_find_entry()
- wdb_metadata_insert_entry()
- wdb_metadata_update_entry()

### wdb_rootcheck.c

- wdb_delete_pm_all()
- wdb_delete_pm()

### wdb.c

- wdb_agent_loc2name()

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

### CODE REVIEW

File: src/wazuh_db/wdb.h [hunk 1]
```diff
@@ -1985,16 +1915,6 @@ int wdb_parse_task_delete_old(wdb_t* wdb, const cJSON *parameters, char* output)
  */
  int wdb_parse_agents_remove_vuln_cves(wdb_t* wdb, char* input, char* output);
```
Comment (pereyra-m) [inline src/wazuh_db/wdb.h:1916]: The doxygen for this method should be updated

**Review by pereyra-m — CHANGES_REQUESTED** [review]
Great job!

Just two minor comments

File: src/wazuh_db/wdb_integrity.c [hunk 2]
```diff
@@ -488,46 +488,6 @@ int wdbi_get_last_manager_checksum(wdb_t *wdb, wdb_component_t component, os_sha
     return result;
 }
 
-// Calculates SHA1 hash from a NULL terminated string array
-int wdbi_array_hash(const char ** strings_to_hash, os_sha1 hexdigest)
```
[the remark this answers is not in the document]
Comment (DProvinciani) [reply src/wazuh_db/wdb_integrity.c:492]: I also think that this helper may be useful in the future. So my suggestion is to keep it there.

File: src/wazuh_db/wdb.h (hunk 1 above)
Comment (MiguelazoDS) [reply src/wazuh_db/wdb.h:1923]: Good catch! thanks

File: src/wazuh_db/wdb_integrity.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (MiguelazoDS) [reply src/wazuh_db/wdb_integrity.c]: Restored

**Review by jnasselle — APPROVED** [review]

**Review by HanesSciarrone — APPROVED** [review]

**Pull request merged**